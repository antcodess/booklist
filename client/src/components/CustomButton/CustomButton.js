import React from 'react';

import './CustomButton.scss';

const CustomButton = ({ children, isGoogleSignIn, isRegularSignIn, ...otherProps }) => {
  return (
    <button 
    className={`custom-button ${isGoogleSignIn ? 'google-sign-in' : ''}${isRegularSignIn ? 'regular-sign-in' : ''}`} 
    {...otherProps}>{ children }</button>
  );
};

export default CustomButton;