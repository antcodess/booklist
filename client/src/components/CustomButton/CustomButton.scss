.regular-sign-in {
  margin-bottom: 5px;
  &:hover {
    background-color: black;
    opacity: 0.9;
  }
}

.google-sign-in {
  background-color: #0080DE;

  &:hover {
    opacity: 0.9;
  }
}

@media only screen and (min-width: 950px) {
  .regular-sign-in {
    width: 40%;
  }
}

@media only screen and (min-width: 1100px) {
  .regular-sign-in, .google-sign-in {
    width: 49%;
  }
}