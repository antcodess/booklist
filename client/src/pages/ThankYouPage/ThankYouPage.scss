.thank-you-page {
  margin: 0 auto;

  .thank-you {
    margin-top: 60px;
    
    .image-container {
      width: 8%;
      margin: 0 auto 10px;
      
      img {
        max-width: 100%;
        height: auto;
      }
    }
  
    h2 {
      font-size: 2rem;
      margin-bottom: 10px;
    }
  
    span {
      font-family: Helvetica, Arial, "Lucida Grande", sans-serif;
    }
  
    .details {
      display: block;
      color: #6A6A6A;
      margin-bottom: 10px;
    }
    
    .custom-button {
      margin: 0 auto;
    }
  }
  
}