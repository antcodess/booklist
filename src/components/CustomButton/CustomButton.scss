.regular-sign-in {
  margin-bottom: 5px;
}

.google-sign-in {
  background-color: #0080DE;

  &:hover {
    background-color: black;
  }
}