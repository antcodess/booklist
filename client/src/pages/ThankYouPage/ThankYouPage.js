import React from 'react';
import { Redirect, withRouter } from 'react-router-dom';

import BreadCrumbNav from '../../components/BreadCrumbNav/BreadCrumbNav';
import CustomButton from '../../components/CustomButton/CustomButton';

import './ThankYouPage.scss';

const ThankYouPage = ({ location, history }) => {
  return (
    <React.Fragment>
      <BreadCrumbNav title='BookList Thanks You'/>
      {
        location.state ?
          location.state.token ?
            <section className='thank-you-page'>
              <div className='thank-you'>
                <div className='image-container'>
                  <img src='https://i.ibb.co/PzKMPjS/thank-you-success.png' alt='success' />
                </div>
                <h2>Thank you for your purchase!</h2>
                <span className='details'>
                  Your order is confirmed.
                </span>
                <CustomButton onClick={() => history.push('/account')}>
                  View Order
                </CustomButton>
              </div>
            </section>
          : <Redirect to='/' />
        : <Redirect to='/' />
      }

    </React.Fragment>
  )
};

export default withRouter(ThankYouPage);