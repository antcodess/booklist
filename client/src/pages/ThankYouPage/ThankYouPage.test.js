import React from 'react';
import { shallow } from 'enzyme';
import { ThankYouPage } from './ThankYouPage';
import CustomButton from '../../components/CustomButton/CustomButton';

describe('ThankYouPage Component', () => {
  let wrapper;
  let wrapper2;
  let wrapper3;
  let mockLocation;
  let mockLocation2;
  let mockLocation3;
  let mockHistory;

  beforeEach(() => {
    mockLocation = {
      state: {
        token: '12819372874824'
      }
    }

    mockLocation2 = {
      state: {}
    };

    mockLocation3 = {};

    mockHistory = { push: jest.fn() };

    const mockProps = {
      location: mockLocation,
      history: mockHistory
    };

    const mockProps2 = {
      location: mockLocation2,
      history: mockHistory
    };

    const mockProps3 = {
      location: mockLocation3,
      history: mockHistory
    };

    wrapper = shallow(<ThankYouPage { ...mockProps } />);
    wrapper2 = shallow(<ThankYouPage { ...mockProps2 } />);
    wrapper3 = shallow(<ThankYouPage { ...mockProps3 } />);
  });

  it('should render ThankYouPage Component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call history.push when CustomButton component is clicked', () => {
    wrapper.find(CustomButton).simulate('click');
    expect(mockHistory.push).toHaveBeenCalled();
  });

  it('should not render thank you page if there is not state on location object', () => {
    expect(wrapper2.exists('.thank-you-page')).toBe(false);
  });
  
  it('should not render thank you page if there is not token', () => {
    expect(wrapper3.exists('.thank-you-page')).toBe(false);
  });
});