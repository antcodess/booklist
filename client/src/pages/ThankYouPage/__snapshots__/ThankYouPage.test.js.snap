// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThankYouPage Component should render ThankYouPage Component 1`] = `ShallowWrapper {}`;
